flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/source/TableStoreSource.java

```diff
@@ -177,11 +181,36 @@ public String asSummaryString() {
 
     @Override
     public Result applyFilters(List<ResolvedExpression> filters) {
+        List<ResolvedExpression> partitionFilters = new ArrayList<>();
+        List<ResolvedExpression> fieldFilters = new ArrayList<>();
+
         if (tableStore.partitioned()) {
-            classifyFilters(filters);
+            classifyFilters(filters, partitionFilters, fieldFilters);
         } else {
             fieldFilters = filters;
         }
+        fieldPredicate =
+                fieldFilters.stream()
+                        .map(PredicateConverter::convert)
+                        .filter(Either::isLeft)
+                        .map(Either::left)
+                        .reduce(And::new)
+                        .orElse(null);
+        Stream<Either<Predicate, ResolvedExpression>> partitionStream =
```

Comment:
I think `Either` is not good to understand. I think we can just classify them in a simple loop. For example:
```
List<Predicate> partitionFilters = new ArrayList<>();
List<Predicate> fieldFilters = new ArrayList<>();
List<ResolvedExpression> notAcceptedFilters = new ArrayList<>();
for (ResolvedExpression filter: filters) {
     if (filter can be a partition filter) {
        ....
     } else ....
}
```